### DISCUSSION TITLE
Explanation of `external_threads` Configuration

### DISCUSSION THREAD

**Discussion by brianwyka** [duckdb/duckdb#11282] [Q&A] [answered]
From the [docs](https://duckdb.org/docs/configuration/overview.html)

> The number of external threads that work on DuckDB tasks.

I'm looking for a bit more clarification.  Is this a way for the API to tell DuckDB it's concurrency?

For example if I am using DuckDb JDBC and I have `10` parallel threads doing work in `memory` database mode, should I `SET external_threads = 10` here? 

Does it make a difference if the connection is shared or duplicated?

Thanks!

**Comment by Tishj** [comment]
external threads are generally only used to indicate to the system that you have spawned threads that serve as worker threads, partaking in the task execution loop, so I don't think that's something you have to worry about here.

It sounds like you have 10 separate in-memory databases running, which all spawn a certain amount of threads for their execution?

This is one of the latest PRs targeting external threads:
<https://github.com/duckdb/duckdb/pull/10548>

**Reply by brianwyka** [reply]
Thanks for the response @Tishj .

I thought the number of worker threads was indicated with `threads` or `worker_threads` configuration.

Example java code:

```java
final ExecutorService threadPool = Executors.newFixedThreadPool(10);

final Properties connectionProperties = new Properties();
connectionProperties.setProperty("access_mode", "READ_WRITE");

try (final Connection connection = DriverManager.getConnection("jdbc:duckdb:", connectionProperties)) {

    for (int 1 = 1,; i<=10; i++) {
        threadPool.submit(() -> {
            try (final Statement statement = connection.createStatement()) { // shared connection used here.
                statement.executeQuery("SELECT 1");
            }
        });
    }
}    
```

let's assume that the number of cores on the machine is `8`, so DuckDB defaults it's internal worker threads to `8`.

**Reply by Tishj** [reply, marked as answer]
Precisely, "internal" worker threads.
That does not include the external threads

Your example does not need to worry about the external_threads setting

**Reply by brianwyka** [reply]
Thanks!
